service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/post-startup.sh

```diff
@@ -48,33 +48,63 @@ set -o xtrace
 # This post startup script is not run by the same user.
 readonly JUPYTER_USER="jupyter"
 
+# Create an alias for cases when we need to run a shell command as the jupyter user.
+# Note that we deliberately use "bash -l" instead of "sh" in order to get bash (instead of dash)
+# and to pick up changes to the .bash_profile.
+#
+# This is intentionally not a Bash function, as that can suppress error propagation.
+# This is intentionally not a Bash alias as they are not supported in shell scripts.
+readonly RUN_AS_JUPYTER_USER="sudo -u "${JUPYTER_USER}" bash -l -c"
```

Comment:
I think this static code analysis is new-- how cool that it's catching easy-to-make (and miss) errors like this.